crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -473,17 +483,21 @@ impl WidgetMachine {
         self.send_to_widget_request(RequestCapabilities {})
             // TODO: Each request can actually fail here, take this into an account.
             .then(|response, machine| {
-                let desired_capabilities = response.capabilities;
+                let requested = response.capabilities;
                 machine
                     .send_matrix_driver_request(AcquireCapabilities {
-                        desired_capabilities: desired_capabilities.clone(),
+                        desired_capabilities: requested.clone(),
                     })
-                    .then(|granted_capabilities, machine| {
-                        machine.capabilities =
-                            CapabilitiesState::Negotiated(granted_capabilities.clone());
+                    .then(|result, machine| {
+                        let approved = result.unwrap_or_else(|e| {
+                            error!("Acquiring capabilities failed: {e}");
+                            Default::default()
```

Comment:
Could this type be expanded? It's not clear from the context if it's a boolean or another type. Probably not a boolean since it's cloned later, so even better to state what it is very clearly here.